lib/sharded-adapter.ts

```diff
@@ -28,9 +28,17 @@ export interface ShardedRedisAdapterOptions {
    * - a lot of connected clients would mean a lot of subscription/unsubscription
    * - the Socket ID attribute is ephemeral
    *
+   * This can be changed by setting "dynamicPrivateChannels" to "true".
+   *
    * @default "dynamic"
    */
   subscriptionMode?: "static" | "dynamic";
```

Comment:
Do you think it would make sense to add a new `subscriptionMode`, something like `dynamic-private` or `dynamic++`?

Reply:
Certainly an option if you think that's better, I considered adding this as `dynamic-private` too.

Reply:
I changed it to `dynamic-private` and rebased on top of the latest changes.

Reply:
... and managed to break it in the process, but it's fixed now.